src/metabase/api/common/openapi.clj

```diff
@@ -3,90 +3,79 @@
-                                      (for [[k v] props]
-                                        [k
-                                         (cond
-                                           (and (:oneOf v)
-                                                (= (second (:oneOf v)) {:type "null"}))
-                                           (do
-                                             (swap! not-required conj k)
-                                             (merge (first (:oneOf v))
-                                                    (select-keys v [:description :default])))
-
-                                           (= (select-keys v (keys file-schema)) file-schema)
-                                           ;; I got this from StackOverflow and docs are not in agreement, but RapiDoc
-                                           ;; shows file input, so... :)
-                                           (merge {:type "string" :format "binary"}
-                                                  (select-keys v [:description :default]))
-
-                                           (= (:enum v) ["true" "false" true false])
-                                           (-> (dissoc v :enum) (assoc :type "boolean"))
-
-                                           :else
-                                           v)]))))
-          (assoc :required (vec (remove @not-required required)))))))
+      (swap! *definitions* merge (:definitions jss)))
+    (dissoc jss :definitions)))
+
+(defn- merge-required [schema]
+  (let [optional? (set (keep (fn [[k v]] (when (:optional v) k))
+                             (:properties schema)))]
+    (-> schema
+        (m/update-existing :required #(vec (remove optional? %)))
+        (m/update-existing :properties #(update-vals % (fn [v] (dissoc v :optional)))))))
+
+(let [file-schema (mjs/transform ms/File)]
+  (defn- fix-json-schema
+    "Clean-up JSON schema to make it more understandable for OpenAPI tools.
+
+    Returns a new schema WITH an indicator if it's *NOT* required.
+
+    NOTE: maybe instead of fixing it up later we should re-work Malli's json-schema transformation into a way we want it to be?"
+    [schema]
+    (cond
+      (and (:oneOf schema)
+           (= (second (:oneOf schema)) {:type "null"}))
+      (recur (-> (first (:oneOf schema))
+                 (merge (select-keys schema [:description :default]))
+                 (assoc :optional true)))
+
+      (:allOf schema)
+      (cond
+        (= {} (first (:allOf schema)))  (recur (merge (second (:allOf schema))
+                                                      (select-keys schema [:description :default])))
+        (= {} (second (:allOf schema))) (recur (merge (first (:allOf schema))
+                                                      (select-keys schema [:description :default])))
+        :else                           (update schema :allOf (partial mapv fix-json-schema)))
+
+      (= (:enum schema) (mc/children ms/BooleanValue))
```

Comment:
this should be a `set`

Reply:
why? :)

Reply:
do we care about the order tho?

Reply:
nope, but we don't care about sets either while this works; but I figured out a better way to make this work :-)